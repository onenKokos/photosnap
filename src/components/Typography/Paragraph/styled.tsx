import styled from 'styled-components';
import theme from '@Common/theme';

export const StyledParagraph = styled.p<{ theme: 'light' | 'dark' }>`
  font-size: 1rem;
  line-height: 1.5rem;
  margin-bottom: 1.5rem;
  color: ${(props) =>
    props.theme === 'light' ? `${theme.colors.muted}` : `${theme.colors.primaryDark}`};
`;
