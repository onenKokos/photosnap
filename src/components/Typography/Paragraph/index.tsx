import React from 'react';
import { StyledParagraph } from './styled';
export interface ParagraphProps {
  text: string;
  theme: 'light' | 'dark';
}

const Paragraph = ({ text, theme }: ParagraphProps) => (
  <StyledParagraph data-testid="paragraph" theme={theme}>
    {text}
  </StyledParagraph>
);

export default Paragraph;
