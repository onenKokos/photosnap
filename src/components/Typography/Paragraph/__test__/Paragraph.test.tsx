import React from 'react';
import { render } from '@testing-library/react';
import Paragraph, { ParagraphProps } from '../';

const renderParagraph = (props: Partial<ParagraphProps> = {}) => {
  const defaultProps: ParagraphProps = {
    text: 'test',
    theme: 'dark',
  };

  return render(<Paragraph {...defaultProps} {...props} />);
};

describe('[Component]: Paragraph', () => {
  it('Should render correctly', () => {
    const renderer = renderParagraph();
    expect(renderer.container).toMatchSnapshot();
  });

  test('Should contain correct text', async () => {
    const { findByTestId } = renderParagraph({
      text: 'test',
    });

    const paragraph = await findByTestId('paragraph');

    expect(paragraph).toContainHTML('test');
  });
});
