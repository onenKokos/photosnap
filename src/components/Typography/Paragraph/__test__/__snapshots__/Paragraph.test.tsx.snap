// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Paragraph Should render correctly 1`] = `
<div>
  <p
    class="sc-bdfBwQ hKiDRp"
    data-testid="paragraph"
  >
    test
  </p>
</div>
`;
